tests/main/interfaces-joystick/task.yaml

```diff
@@ -0,0 +1,57 @@
+summary: Ensure that the joystick interface works.
+
+details: |
+    The joystick interface allows reading and writing to joystick devices.
+
+prepare: |
+    . "$TESTSLIB/snaps.sh"
+    install_local test-snapd-joystick
+
+    . "$TESTSLIB/files.sh"
+    ensure_file_exists_backup_real /dev/input/js31
+    ensure_file_exists_backup_real /run/udev/data/c13:31
+    ensure_file_exists_backup_real /run/udev/data/c13:67
+    ensure_file_exists_backup_real /dev/input/event67
+
+restore: |
+    rm -f call.error
+
+    . "$TESTSLIB/files.sh"
+    clean_file /dev/input/js31
+    clean_file /run/udev/data/c13:31
+    clean_file /run/udev/data/c13:67
+    clean_file /dev/input/event67
+
+execute: |
+    echo "The interface is not connected by default"
+    snap interfaces -i joystick | MATCH "\- +test-snapd-joystick:joystick"
+
+    echo "When the interface is connected"
+    snap connect test-snapd-joystick:joystick
+
+    echo "Then the snap is able to access the device input for the old interface"
+    test-snapd-joystick.sh -c "echo test >> /dev/input/js31"
+    test-snapd-joystick.sh -c "cat /run/udev/data/c13:31"
+
+    echo "Then the snap is able to access the device input for the new interface"
+    test-snapd-joystick.sh -c "cat /run/udev/data/c13:67"
+    test-snapd-joystick.sh -c "echo test >> /dev/input/event67"
+
+    capabilities="$(find /sys/devices/ -type d -name capabilities | grep -E "/sys/devices/.*/input[0-9].*/capabilities" | head -n1)"
+    if [ ! -z $capabilities ]; then
+        test-snapd-joystick.sh -c "ls "$capabilities""
+    fi
+
+    if [ "$(snap debug confinement)" = partial ] ; then
```

Comment:
Is this really constrained by partial confinement? In partial confinement we still have the device cgroup which, AFAIK, does all the heavy lifting here.

Reply:
well, it is failing when we run without that check:
+ echo 'When the plug is disconnected'
When the plug is disconnected
+ snap disconnect test-snapd-joystick:joystick
+ echo 'Then the snap is not able to read the input device'
Then the snap is not able to read the input device
+ test-snapd-joystick.sh -c 'cat /dev/input/js31'
test
+ echo 'Expected permission error accessing to input device'
Expected permission error accessing to input device
+ exit 1

Reply:
It fails because device cgroup probably needs a device, while here we have just a file.